projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -0,0 +1,152 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from '../base-value.component';
+import {CreateLinkValue, ReadLinkValue, ReadResource, UpdateLinkValue, KnoraApiConnection} from '@knora/api';
+import {Subscription} from 'rxjs';
+import {AbstractControl, FormBuilder, FormControl, FormGroup, ValidatorFn} from '@angular/forms';
+import {KnoraApiConnectionToken} from '../../../core';
+
+@Component({
+  selector: 'kui-link-value',
+  templateUrl: './link-value.component.html',
+  styleUrls: ['./link-value.component.scss']
+})
+export class LinkValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() displayValue?: ReadLinkValue;
+  @Input() parentResource: ReadResource;
+  @Input() propType: string;
+  resources: ReadResource[];
+  restrictToResourceClass: string;
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+  labelChangesSubscription: Subscription;
+  // label cannot contain logical operations of lucene index
+  customValidators = [];
```

Comment:
the use case is here is more complex because the value of the `FormControl` is a string / label as long as the user types and an instance of `ReadResource` once the user selects from the list.

Reply:
> the use case is here is more complex because the value of the `FormControl` is a string / label as long as the user types and an instance of `ReadResource` once the user selects from the list.

If I understand what you suggest correctly, you mean there should be a validator to check that a chosen option is an instance of ReadResource. However, in "update", "create", and "search" modes,  the dropdown menu shows the options for the user to choose from. These options are always instances of ReadResource. The user does not have any other choice than choosing one item from the menu. Hence, an option selected by the user is always an instance of ReadResource. That would mean that such a double validation would not actually be necessary.

Reply:
> However, in "update", "create", and "search" modes, the dropdown menu shows the options for the user to choose from. These options are always instances of ReadResource. The user does not have any other choice than choosing one item from the menu. Hence, an option selected by the user is always an instance of ReadResource. That would mean that such a double validation would not actually be necessary.

Yes, but as long as the user types, the `FormControl`'s value is just a string (the label). So this is to make sure that the user selected an option from the list.

Reply:
the "save" button is inactive unless an option is chosen from the dropdown menu.

Reply:
this is true for update mode because the check is integrated into standardValidatorFunc

please try with create

Reply:
the standard behavior is not enough here since `Validators.required` checks that the value is not `null` (https://angular.io/api/forms/Validators#required). In the case to the link value, a string is not valid, only a `ReadResource`.

Reply:
do we have already a component for create mode?

Reply:
no, not yet. But you test it by directly adding link value component to `app.component.html`. I think @mdelez will wok on a create component soon.

Reply:
or you can just try it out in your spec

Reply:
I am trying out in spec for a while now. It would have been nice to have the component.

Reply:
yes! Now we can make it a user story :-)

Reply:
@SepidehAlassi I can start to work on the create component tomorrow